src/common/fetch/download-binary.injectable.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import type { RequestInit, Response } from "node-fetch";
+import type { AsyncResult } from "../utils/async-result";
+import fetchInjectable from "./fetch.injectable";
+
+export interface DownloadBinaryOptions {
+  signal?: AbortSignal | null | undefined;
+}
+
+export type DownloadBinary = (url: string, opts?: DownloadBinaryOptions) => Promise<AsyncResult<Buffer, string>>;
+
+const downloadBinaryInjectable = getInjectable({
+  id: "download-binary",
+  instantiate: (di): DownloadBinary => {
+    const fetch = di.inject(fetchInjectable);
+
+    return async (url, opts) => {
+      let result: Response;
+
+      try {
+        // TODO: upgrade node-fetch once we switch to ESM
+        result = await fetch(url, opts as RequestInit);
+      } catch (error) {
+        return {
```

Comment:
To answer both this and your other review comment, this style of programming is called "Errors as data" instead of "Errors as exceptions". That means we have started using the more functional programming approach.

This allows us to better encode the error states into the program, to explicitly change the errors we expect into just data, and remove the need to use try/catch in business logic.

Furthermore, we have moved to using a single type definition `AsyncResult` 

```ts
export type AsyncResult<Response, Error = string> =
  | (
    Response extends void
    ? { callWasSuccessful: true; response?: undefined }
    : { callWasSuccessful: true; response: Response }
  )
  | { callWasSuccessful: false; error: Error };
```

So that the error handling code is consistent throughout.